File: ReviewCpp/ClassExample/Point.cc
#include "Point.h"


Point::Point( double ix, double iy) { 
  x_=ix;y_=iy;
}

Point::~Point(){}

void Point::print() const {
  std::cout << "(" << x_ << "," << y_ << ")" << std::endl;
};

double Point::x() const { return x_;}
double Point::y() const { return y_;}

Point Point::operator+( Point const & right ) const {
  Point retval( x_ + right.x_, y_ + right.y_ );
  return retval;
}

Point Point::operator-( Point const & right ) const {
  Point retval( x_ - right.x_, y_ - right.y_ );
  return retval;
}

Point & Point::operator+=( Point const & right )  {
  x_ += right.x_; y_ += right.y_ ;
  return *this;
}

Point & Point::operator-=( Point const & right )  {
  x_ -= right.x_; y_ -= right.y_ ;
  return *this;
}

bool Point::input( std::istream & in ) {

  std::string line;   
  std::getline( in, line, ',');   
  x_ = std::atof( line.c_str() ); 
  std::getline( in, line );   
  y_ = std::atof( line.c_str() );       
  if ( line == "") {      
    return false;  
  } 
  else {
    return true; 
  }
}

File: ReviewPython/Point.py
class Point:
    x_ = 0.
    y_ = 0.

    def __init__( self, x=0.,y=0.) :
        self.x_ = x
        self.y_ = y

    def __add__( self, right ):
        return Point( self.x_ + right.x_, self.y_ + right.y_ )

    def __sub__( self, right ):
        return Point( self.x_ - right.x_, self.y_ - right.y_ )

    def __str__(self ):
        return '(' + str( self.x_ ) + ',' + str(self.y_) + ')'

    def input(self, line):
        self.x_, self.y_ = [s.strip() for s in line.split(",")]

    
